scripts/otp.js

```diff
@@ -0,0 +1,43 @@
+// Description:
+//   one time password for authentication
+//   it is tested on slack and heroku
+// Commands:
+//   hubot otp - get otp
+
+var timeout = parseInt(process.env.HUBOT_OTP_TIMEOUT) || 30;
+var length = parseInt(process.env.HUBOT_OTP_LENGTH) || 6;
+
+module.exports = function(robot){
+  robot.router.post("/hubot/otp/:name", function(req, res){
+    var token = req.body.token;
+    var name = req.params.name;
+
+    if (robot.brain.get(name + ".otp") == token){
+      res.status(200).jsonp({});
+      robot.brain.set(name + ".otp", null);
+    } else {
+      res.status(401).jsonp({});
+    }
+  });
+
+  robot.respond(/otp/i, otp);
+  function otp(res){
+    var token = generateToken();
+    var name = res.envelope.user.name;
+    robot.messageRoom(name, "Your token is <" + token + ">. It will be invalid in " + timeout + " seconds.");
+    robot.brain.set(name + ".otp", token);
```

Comment:
Stored tokens should be encrypted by a cryptographic hash function. (e.g. SHA-2)